packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -446,10 +446,15 @@ export class DumbTriggerManager extends Component {
         uri: get(konnector, 'vendor_link')
       })
 
-      if (ciphers.length === 0) {
-        this.showAccountForm()
-      } else {
-        this.showCiphersList(ciphers)
+      // When we fetched the ciphers while the user is already on the account
+      // form step, we don't want to automatically redirect him on the ciphers
+      // list
+      if (this.state.step !== 'accountForm') {
+        if (ciphers.length === 0) {
+          this.showAccountForm()
+        } else {
+          this.showCiphersList(ciphers)
+        }
```

Comment:
This is a good use case for a more declarative code ? :D

Reply:
I'm not sure I understand your point. Do you mean it should be handled via components rendered?

Reply:
Just discussed it with @ptbrowne. He is going to refactored all this in a more declarative way while adding 2FA feature for banking konnectors. Since here I added missing test cases, we merge this PR so he has the tests to allow him to refactor easily.